engine/src/main/scala/cromwell/jobstore/FilesystemJobStoreDatabase.scala

```diff
@@ -0,0 +1,75 @@
+package cromwell.jobstore
+
+import java.io.File
+import java.nio.file.{Files, Paths}
+
+import cromwell.core.ExecutionIndex._
+import cromwell.core.WorkflowId
+import org.apache.commons.io.FileUtils
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+case object FilesystemJobStoreDatabase extends JobStoreDatabase {
+
+  val separator = File.separator
+  val fsroot = Paths.get("JobStore").toAbsolutePath
+
+  implicit class EnhancedJobStoreKey(val jobStoreKey: JobStoreKey) extends AnyVal {
+    def fileDir = Paths.get(s"${fsroot.toString}$separator${jobStoreKey.workflowId}").toAbsolutePath
+    def filePath = Paths.get(s"$fileDir$separator${jobStoreKey.callFqn}_index${jobStoreKey.index.fromIndex}_attempt${jobStoreKey.attempt}")
+  }
+
+  override def writeToDatabase(jobCompletions: Map[JobStoreKey, JobResult], workflowCompletions: List[WorkflowId])(implicit ec: ExecutionContext): Future[Unit] = {
+    for {
+      _ <- writeJobResultsSerially(jobCompletions)
+      _ <- writeWorkflowResultsSerially(workflowCompletions)
+    } yield ()
+  }
+
+  private def writeJobResultsSerially(jobCompletions: Map[JobStoreKey, JobResult])(implicit ec: ExecutionContext): Future[Unit] = {
+
+    def folderFunction(current: Future[Unit], nextPair: (JobStoreKey, JobResult)): Future[Unit] = {
+      val (jobStoreKey, jobResult) = nextPair
+      current flatMap { _ => writeJobResult(jobStoreKey, jobResult) }
+    }
+    jobCompletions.foldLeft(Future.successful(()))(folderFunction)
+  }
+
+  private def writeJobResult(jobStoreKey: JobStoreKey, jobResult: JobResult)(implicit ec: ExecutionContext): Future[Unit] = {
+    Future {
+      import JobResultJsonFormatter._
+      val fileDir = jobStoreKey.fileDir
+      if (Files.notExists(fileDir)) Files.createDirectories(fileDir)
+      FileUtils.writeStringToFile(jobStoreKey.filePath.toFile, jobResult.toJson.toString)
+    }
+  }
+
+  private def writeWorkflowResultsSerially(workflowCompletions: List[WorkflowId])(implicit ec: ExecutionContext): Future[Unit] = {
+
+    def folderFunction(current: Future[Unit], nextId: WorkflowId): Future[Unit] = current flatMap { _ => clearWorkflowResults(nextId) }
+
+    workflowCompletions.foldLeft(Future.successful(()))(folderFunction)
+  }
+
+  private def clearWorkflowResults(workflowId: WorkflowId)(implicit ec: ExecutionContext): Future[Unit] = {
+    Future {
+      val path = Paths.get(s"${fsroot.toString}$separator$workflowId").toAbsolutePath
+      if (Files.exists(path)) {
+        path.toFile.listFiles foreach { x =>
+          System.out.println(s"Nuking $path from orbit!")
```

Comment:
I know this is one of mine but... I still probably wouldn't advise leaving it in the production system